src/sage/schemes/curves/zariski_vankampen.py

```diff
@@ -929,82 +954,84 @@ def geometric_basis(G, E, EC0, p, dual_graph):
-          A vertex at (-2, -2)],
-         [A vertex at (-2, -2),
-          A vertex at (-1/2, -1/2),
-          A vertex at (-1/2, 1/2),
-          A vertex at (1/2, 1/2),
-          A vertex at (2, 2),
-          A vertex at (-2, 2),
-          A vertex at (-1/2, 1/2),
-          A vertex at (-1/2, -1/2),
-          A vertex at (-2, -2)],
-         [A vertex at (-2, -2),
-          A vertex at (-1/2, -1/2),
-          A vertex at (-1/2, 1/2),
-          A vertex at (-2, 2),
-          A vertex at (-2, -2)]]
+        sage: from sage.schemes.curves.zariski_vankampen import geometric_basis, corrected_voronoi_diagram, voronoi_cells
+        sage: points = (0, -1, I, 1, -I)
+        sage: V = corrected_voronoi_diagram(points)
+        sage: G, E, p, EC, DG, VR = voronoi_cells(V, vertical_lines=[0, 1, 2, 3, 4])
+        sage: gb, vd = geometric_basis(G, E, EC, p, DG, VR)
+        sage: gb
+        [[A vertex at (5/2, -5/2), A vertex at (5/2, 5/2), A vertex at (-5/2, 5/2),
+          A vertex at (-1/2, 1/2), A vertex at (-1/2, -1/2), A vertex at (1/2, -1/2),
+          A vertex at (1/2, 1/2), A vertex at (-1/2, 1/2), A vertex at (-5/2, 5/2),
+          A vertex at (5/2, 5/2), A vertex at (5/2, -5/2)],
+         [A vertex at (5/2, -5/2), A vertex at (5/2, 5/2), A vertex at (-5/2, 5/2),
+          A vertex at (-1/2, 1/2), A vertex at (1/2, 1/2), A vertex at (5/2, 5/2),
+          A vertex at (5/2, -5/2)],
+         [A vertex at (5/2, -5/2), A vertex at (5/2, 5/2), A vertex at (1/2, 1/2),
+          A vertex at (1/2, -1/2), A vertex at (5/2, -5/2)], [A vertex at (5/2, -5/2),
+          A vertex at (1/2, -1/2), A vertex at (-1/2, -1/2), A vertex at (-1/2, 1/2),
+          A vertex at (-5/2, 5/2), A vertex at (-5/2, -5/2), A vertex at (-1/2, -1/2),
+          A vertex at (1/2, -1/2), A vertex at (5/2, -5/2)],
+         [A vertex at (5/2, -5/2), A vertex at (1/2, -1/2), A vertex at (-1/2, -1/2),
+          A vertex at (-5/2, -5/2), A vertex at (5/2, -5/2)]]
+        sage: vd
+        {0: 0, 1: 3, 2: 1, 3: 2, 4: 4}
     """
     i = EC0.index(p)
-    EC = EC0[i:-1] + EC0[:i + 1]   # A counterclockwise eulerian circuit on the boundary, starting and ending at p
+    EC = EC0[i:-1] + EC0[:i + 1]
+    # A counterclockwise eulerian circuit on the boundary,
+    # starting and ending at p
     if G.size() == E.size():
         if E.is_cycle():
-            return [EC]
-    InternalEdges = [_ for _ in G.edges(sort=True) if _ not in E.edges(sort=True)]
+            j = list(dual_graph.vertices())[0][0]
+            if j in vertical_regions:
+                vd = {j: 0}
+            else:
+                vd = dict()
+            return [EC], vd
+    InternalEdges = [_ for _ in G.edges(sort=True) if _ not in
+                     E.edges(sort=True)]
     InternalVertices = [v for e in InternalEdges for v in e[:2]]
-    Internal = G.subgraph(vertices=InternalVertices, edges=InternalEdges)
+    Intern = G.subgraph(vertices=InternalVertices, edges=InternalEdges)
```

Comment:
Why this change? This now makes it seem like we've hired someone to do this, so it is less descriptive to me. It also creates a bunch of unnecessary changes for this PR.